codegen/type_converter.go

```diff
@@ -804,9 +896,13 @@ func checkOptionalNil(
 	for k := range uninitialized {
 		keys = append(keys, k)
 	}
+
```

Comment:
I think we have introduced to many piece of information here to simply identify some name in different layers, it won't be too long before type_converter is completely unmaintainable